libbeat/registry/store_test.go

```diff
@@ -0,0 +1,239 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package registry
+
+import (
+	"errors"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestStoreBeginTx(t *testing.T) {
+	t.Run("closed", testStoreBeginTxClosed)
+	t.Run("run tx", testStoreWithTx)
+}
+
+func testStoreBeginTxClosed(t *testing.T) {
+	makeTestStore := func(t *testing.T, ms *mockStore) *Store {
+		mr := newMockRegistry()
+		mr.OnAccess("test").Once().Return(ms, nil)
+
+		reg := NewRegistry(mr)
+
+		s, err := reg.Get("test")
+		require.NoError(t, err)
+
+		ms.OnClose().Return(nil)
+		require.NoError(t, s.Close())
+		return s
+	}
+
+	t.Run("begin readonly", func(t *testing.T) {
+		ms := newMockStore()
+		store := makeTestStore(t, ms)
+
+		_, err := store.Begin(true)
+		assert.Error(t, err)
+
+		ms.AssertExpectations(t)
+		ms.AssertNotCalled(t, "Begin")
+	})
+
+	t.Run("begin rw", func(t *testing.T) {
+		ms := newMockStore()
+		store := makeTestStore(t, ms)
+
+		_, err := store.Begin(false)
+		assert.Error(t, err)
+
+		ms.AssertExpectations(t)
+		ms.AssertNotCalled(t, "Begin")
+	})
+
+	t.Run("update fails", func(t *testing.T) {
+		ms := newMockStore()
+		store := makeTestStore(t, ms)
+
+		err := store.Update(func(tx *Tx) error { return nil })
+		assert.Error(t, err)
+
+		ms.AssertExpectations(t)
+		ms.AssertNotCalled(t, "Begin")
+	})
+
+	t.Run("view fails", func(t *testing.T) {
+		ms := newMockStore()
+		store := makeTestStore(t, ms)
+
+		err := store.Update(func(tx *Tx) error { return nil })
+		assert.Error(t, err)
+
+		ms.AssertExpectations(t)
+		ms.AssertNotCalled(t, "Begin")
+	})
+
+	t.Run("GC fails", func(t *testing.T) {
+		ms := newMockStore()
+		store := makeTestStore(t, ms)
+
+		err := store.Update(func(tx *Tx) error { return nil })
+		assert.Error(t, err)
+
+		ms.AssertExpectations(t)
+		ms.AssertNotCalled(t, "Begin")
+	})
+}
```

Comment:
Same code as the previous tests?

Reply:
ah, some leftover when I still have had a gc method, which I removed. Will remove the test for now.